src/Compilers/CSharp/Portable/Binder/Binder.cs

```diff
@@ -682,6 +682,17 @@ private static ThreeState ReportDiagnosticsIfObsoleteInternal(DiagnosticBag diag
         internal enum BindValueKind : byte
         {
             /// <summary>
+            /// Same as <see cref="RValue"/>, but in context where there is a target type for it.
```

Comment:
Please write up a summary describing the kind of specification change you are contemplating (i.e. the "infrastructure" that would be required in the C# language specification, and how and where it would be used). We should have LDM approval before integrating this (as opposed to the "traditional" approach of allowing a tuple expression to be of its natural type).